### ISSUE TITLE
Inference on different dataset, and further transfer learning

### ISSUE THREAD

**Issue by toefl-girl-hizz** [pmixer/SASRec.pytorch#45]
Dear contributors,

Thank you so much for your contribution.  Lately I've been experimenting with your code. I tried both training and only inference. However, when I try to inference Video.txt on model trained with ml-1m(the ones provided in ml-1m_default), an error is returned. Just like the picture shown below.

![image](https://github.com/user-attachments/assets/14003a81-6d43-44d6-8775-0fc3dd0b4368)

1. Sorry if I miss anything, but isn't the SASRec model architecture be the same for these 2 datasets? When I tried inferencing ml-1m with model trained with ml-1m, there is nothing wrong with it. What may be wrong here? Or what exactly architecture or layer size is different between a model trained on 2 different datasets?
2. Also, I want to finetune the model trained with ml-1m on Video. What can I do to fix this?
I noticed that the number of users and number of items differ in these 2 datasets, but I could not figure if this is the reason.

Please help, thanks to anyone who has answers

**Comment by pmixer** [comment]
The static linear embedding table(`torch.nn.Embedding`) would prevent from achieving what you're trying to do.

Modules like https://github.com/tensorflow/recommenders-addons may provide the capability needed.

Btw, it do not make sense to me of simply trying to do inference on another dataset.

**Comment by toefl-girl-hizz** [comment]
Thanks for your response. I want to do inference on another dataset because I want to know how well the pre-trained model perform before finetuning. I need to figure out a better way to do finetuning or does my original thoughts make sense.

**Comment by pmixer** [comment]
> Thanks for your response. I want to do inference on another dataset because I want to know how well the pre-trained model perform before finetuning. I need to figure out a better way to do finetuning or does my original thoughts make sense.

So, try replacing torch.nn.Embedding with dynamic embedding table implementation by your own pls.

I don't think pre-training/finetuning are good fits for the scenario, as even dropout rates got tuned for different datasets.

But it would still be interesting if you could get some experimental results and insights from it.

Still, its not LLM, pls check cross-market recommendation https://xmrec.github.io/ if hope to try sth like transfer learning.

Lastly, personally, I agree with `A lot of $$$, but not generalizable beyond a company due to reliance on counting/graph/collaborative data.` as claimed in https://big-ann-benchmarks.com/neurips23_slides/ANNS_for_recommendation_systems_Yury.pdf, as personalization sounds like anti-generalization sometimes.
